kroxylicious/src/main/java/io/kroxylicious/proxy/internal/filter/FilterContributorManager.java

```diff
@@ -7,27 +7,37 @@
 
 import java.util.Iterator;
 import java.util.ServiceLoader;
+import java.util.function.Supplier;
 
 import io.kroxylicious.proxy.config.BaseConfig;
 import io.kroxylicious.proxy.filter.Filter;
 import io.kroxylicious.proxy.filter.FilterContributor;
 
 public class FilterContributorManager {
 
-    private static final FilterContributorManager INSTANCE = new FilterContributorManager();
+    public static final FilterContributorManager INSTANCE = new FilterContributorManager(new Supplier<>() {
 
-    private final ServiceLoader<FilterContributor> contributors;
+        private final ServiceLoader<FilterContributor> serviceLoader = ServiceLoader.load(FilterContributor.class);
 
-    private FilterContributorManager() {
-        this.contributors = ServiceLoader.load(FilterContributor.class);
+        @Override
+        public Iterator<FilterContributor> get() {
+            return serviceLoader.iterator();
+        }
+    });
+
+    private final Supplier<Iterator<FilterContributor>> contributors;
```

Comment:
`Iterable<FilterContributor>` might work here too, might save having to write an anonymous inner class